UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Gas.cs

```diff
@@ -36,33 +37,45 @@ public static class Gas
 	[Serializable]
 	public class GasData
 	{
-		//Used for quick iteration
+		//For editor serialisation, used to fill the GasesBag if necessary
 		public List<GasValues> GasesArray = new List<GasValues>();
 
+		//Used for quick iteration
+		public ConcurrentBag<GasValues> Gases = new ConcurrentBag<GasValues>();
+
 		//Used for fast look up for specific gases
 		public Dictionary<int, GasValues> GasesDict = new Dictionary<int, GasValues>();
 
 		public void RegenerateDict()
 		{
 			GasesDict.Clear();
 
-			for (int i = 0; i < GasesArray.Count; i++)
+			foreach (var gasData in Gases)
 			{
-				var value = GasesArray[i];
-				GasesDict.Add(value.GasSO, value);
+				GasesDict.Add(gasData.GasSO, gasData);
 			}
 		}
 
 		public void Clear()
 		{
-			for (int i = 0; i < GasesArray.Count; i++)
+			foreach (var gasData in Gases)
 			{
-				GasesArray[i].Pool();
+				gasData.Pool();
 			}
 
-			GasesArray.Clear();
+			//TODO Unity 2021.2 has .NET standard 2.1 which has a .Clear() method
+			Gases = new ConcurrentBag<GasValues>();
```

Comment:
This also GC's see comment, could make a pool in the mean time? or can do:

`
while (Gases.IsEmpty == false)
			{
				Gases.TryTake(out _);
			}
`